### PR TITLE
bench/test: add tests for the official lib of arrow.

### PR DESCRIPTION
**Pull request by AndreMouche**
## What have you changed? (mandatory)

Add bench tests for the official lib of `arrow`, it seems the official lib behaves similarly to the implementation of `tidb` in iterating the rows, while behaves worse to build the chunk. Since we only build chunk and do not support compute with arrow in `tikv`, we will still use `tidb`'s implementation now. Attention: we may consider to change the `arrow` lib to official lib once we support computation with `arrow` 

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

bench tests

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Benchmark result if necessary (optional)

```
test coprocessor::codec::chunk::bench_chunk_build_offical            ... bench:      48,382 ns/iter (+/- 4,548)
test coprocessor::codec::chunk::bench_chunk_build_tidb               ... bench:      24,833 ns/iter (+/- 2,955)
test coprocessor::codec::chunk::bench_chunk_iter_offical             ... bench:      27,042 ns/iter (+/- 1,714)
test coprocessor::codec::chunk::bench_chunk_iter_tidb                ... bench:      27,727 ns/iter (+/- 2,146)
```

@breeswish @zz-jason PTAL

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -75,6 +75,7 @@ crossbeam = "0.2"
 fxhash = "0.2"
 derive_more = "0.11.0"
 num = "0.2.0"
+arrow = "0.10.0"
```
Comment (siddontang) [inline Cargo.toml]: move this to `[dev-dependencies]`

**Comment by siddontang** [comment]
em, why the official iterator is much better than ours...

**Comment by AndreMouche** [comment]
Sorry, I've forgotten to check null in the test for the official lib, the iterator performance is similar then. I will update the tests later. @siddontang

**Comment by breeswish** [comment]
nice job. Does it mean that we can replace our impl with officials later?

**Comment by siddontang** [comment]
@breeswish 

From the benchmark result, I think we don't need to move to the official implementation. but we can refer its alignment allocation.

**Comment by AndreMouche** [comment]
We won't refer the official lib's alignment allocation in this PR with the following reasons @siddontang :
1. The official lib only supports push data column by column, while we push data row by row since we storage it row by row. If we change it, the performance of the chunk building is not very pretty. 
2. We've already considered the alignment allocation in the current implementation, and it seems not bad.
3. The change may be very large, we'd better implement it in an independent PR if needed.

**Comment by siddontang** [comment]
LGTM

**Comment by AndreMouche** [comment]
@breeswish PTAL

File: benches/benches.rs [hunk 2]
```diff
@@ -17,6 +17,10 @@ extern crate byteorder;
 extern crate crossbeam_channel;
 extern crate kvproto;
 extern crate log;
+
+#[macro_use(slog_o, slog_kv)]
+extern crate slog;
```
Comment (breeswish) [inline benches/benches.rs]: Seems that we don't need to import slog here.

**Comment by AndreMouche** [comment]
@siddontang @zz-jason PTAL

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**